### DISCUSSION TITLE
Putting together a minimal GPU Image (hard)

### DISCUSSION THREAD

**Discussion by samayala22** [runs-on/runs-on#126] [Q&A]
The base DLAMI suggested in the docs for GPU runners is very bloated for my application (I only need drivers + cuda toolkit). I'm trying to replace it by one of 2 ways:

**1. Find an image that only comes with a driver pre-installed.** 

This is surprisingly hard to find. Most community images are pre-filled with unecessary stuff. I found some [NVIDIA images](https://aws.amazon.com/marketplace/pp/prodview-l5sfltz5xpo6e) in the AWS Marketplace category but I CANNOT find the owner id (tried everything).

**2. Install the drivers starting from a standard image** 

Tried [many times](https://github.com/samayala22/tinyvlm/actions/workflows/linux_gpu2.yaml) with various drivers, direct kernel load, docker container (as suggested in https://github.com/runs-on/runs-on/discussions/123) but nothing works. 

This is only part 1 because I would like to use the GPU on a windows image too...

Open to suggestions.

**Comment by crohr** [comment]
Hi @samayala22, forgetting the bloated aspect for a moment, were you able to successfully run your workflows using the base DLAMI? If so, what is the off-putting aspect of keeping them? The boot time, something else?

**Reply by samayala22** [reply]
I should've mentioned it in my post but yes, the base DLAMI works for my workflow. My enquiry was mainly a matter of improving both boot time and having fine grained control over the driver & toolkit versions. 

I may add that I managed to install the driver manually using this [workflow](https://github.com/samayala22/tinyvlm/actions/runs/10656712411/workflow ) on the official Ubuntu image. The one provided by RunsOn didnt work (couldn't load the kernel module for some reason).

**Reply by ruffsl** [reply]
@samayala22 , impressive. You were able to install and use nvidia's drivers and kernel modules without a reboot? I have an example of modifying the packer templates to setup the nvidia's driver and container runtime into the runner AMI:
- https://github.com/runs-on/runner-images-for-aws/pull/5

**Reply by samayala22** [reply]
Yes, no reboot required. Thanks for the example !

**Comment by ruffsl** [comment]
Update:
- https://github.com/runs-on/runner-images-for-aws/pull/14#event-15857583296

**Comment by crohr** [comment]
I'm going to close that discussion in favor of the other one, thanks everyone.
